lib/opam_build.ml

```diff
@@ -116,3 +114,21 @@ let dockerfile ~base ~info ~repo ~variant =
     if Hashtbl.length cache > cache_max_size then Hashtbl.clear cache;
     Hashtbl.add cache key x;
     x
+
+let v (type s) ~docker:(module Docker : S.DOCKER_CONTEXT with type source = s)
+    ~variant ~repo ~analysis (source : s) =
+  let open Current.Syntax in
+  let info =
+    let+ info = analysis in
+    let opam_files = Analyse.Analysis.opam_files info in
+    if opam_files = [] then failwith "No opam files found!";
+    info
+  in
+  let dockerfile =
+    let+ base = Docker.pull ("ocurrent/opam:" ^ variant)
+    and+ repo = repo
+    and+ info = info in
+    dockerfile ~base:(Docker.image_hash base) ~info ~repo ~variant
+  in
+  let build = Docker.build ~label:"" ~dockerfile source in
```

Comment:
I made the `~label` mandatory but in this case it doesn't look nice. Should I add a label here or make the argument optional ?

Reply:
I say make it optional: it's optional in `Current_docker`, and I don't see a particular reason for introducing a difference in this case.

Reply:
Fixed.